app/src/main/java/com/example/drawingApp/ColorPickerUtility.kt

```diff
@@ -0,0 +1,222 @@
+
+            activeBox.setBackgroundColor(intColor)
+        }
+        //called whenever text needs to be updated
+        val updateForceViews = {
+            val pack = onColorTextForceUpdate()
+            setActiveBox(floatArrayOf(pack.hue, pack.saturation, pack.value))
+        }
+        previousBox.setOnClickListener {
+            val hsv = FloatArray(3)
+            Color.colorToHSV(previousColor.toArgb(), hsv)
+            //used to set the color of the active box to the previous color
+            setActiveBox(hsv)
+        }
+        submit.setOnClickListener {
+            onSubmission(
+                Color.HSVToColor(
+                    floatArrayOf(
+                        slider.hue,
+                        square.sat,
+                        square.valu
+                    )
+                ).toColor()
+            )
+            sheet.dismiss()
+        }
+        cancel.setOnClickListener {
+            sheet.dismiss()
+        }
+
+        slider.onHueChange = {
+            square.hue = it
+            square.invalidate()
+            onHueUpdate(it)
+            updateForceViews()
+        }
+        square.onSatOrValChange = { saturation, value ->
+            onSatOrValUpdate(saturation, value)
+            square.sat = saturation
+            square.valu = value
+            updateForceViews()
+        }
+        square.onDown = onViewTouch
+        square.onUp = onViewLift
+        slider.onDown = onViewTouch
+        slider.onUp = onViewLift
+
+        hView.addTextChangedListener {
+            val passedInt =
+                hView.text.toString().toFloatOrNull()?.let { string ->
+                    max(min(360f, string), 0f)
+                }
+            passedInt?.let { hue ->
+                slider.sliderSetHue(hue)
+                onHueUpdate(hue)
+                updateForceViews()
+            }
+        }
+        sView.addTextChangedListener {
+            val saturation = sView.text.toString().toFloatOrNull()
+                ?.let { string -> max(min(1f, string.div(100)), 0f) }
+            val value = vView.text.toString().toFloatOrNull()
+                ?.let { string -> max(min(1f, string.div(100)), 0f) }
+            saturation?.let { sat ->
+                value?.let { valu ->
+                    onSatOrValUpdate(sat, valu)
+                    updateForceViews()
+                }
+            }
+        }
+        vView.addTextChangedListener {
+            val saturation = sView.text.toString().toFloatOrNull()
+                ?.let { string -> max(min(1f, string.div(100)), 0f) }
+            val value = vView.text.toString().toFloatOrNull()
+                ?.let { string -> max(min(1f, string.div(100)), 0f) }
+            saturation?.let { sat ->
+                value?.let { valu ->
+                    onSatOrValUpdate(sat, valu)
+                    updateForceViews()
+                }
+            }
+        }
+        rView.addTextChangedListener {
```

Comment:
Do you have these set in your layout to only allow numeric input?

Reply:
I did not, now they are incorporated